src/main/java/ch/bfh/bti7081/s2018/green/presenters/MedicationPresenter.java

```diff
@@ -16,13 +14,33 @@ public MedicationPresenter(MedicationView view) {
 		this.view = view;
 		this.data = DataContainer.getInstance();
 
-        view.getMedicamentGrid().addSelectionListener(event -> {
-            Notification.show("You have clicked on element");
-        });
+        view.getMedicamentGrid().setSelectionMode(Grid.SelectionMode.SINGLE);
 
-        SingleSelectionModel<Medication> selectionModel = (SingleSelectionModel<Medication>) view.getMedicamentGrid().setSelectionMode(Grid.SelectionMode.SINGLE);
-        selectionModel.addSingleSelectionListener(event -> {
-            Notification.show(selectionModel.getSelectedItem().get().getName());
+        // add edit button with event listener
```

Comment:
Is it possible to add something like double-click instead of a button?

Reply:
Yes and no. In my opinion, some users do not really care of doubleclicking everything. So having a visible element to indicate something can be edited is better.

I don't remember exactly but I think there were problems using double click EventHandler (if there is even one).

We can discuss this for sprint 3 - i like it and some Vaadin examples are made of the same kind.